src/frequenz/sdk/actor/power_distributing/_component_managers/_pv_inverter_manager/_pv_inverter_manager.py

```diff
@@ -121,11 +121,19 @@ async def distribute_power(self, request: Request) -> None:
             raise ValueError(
                 "Cannot distribute power to PV inverters without any inverters"
             )
-        working_components = list(
-            self._component_pool_status_tracker.get_working_components(
-                request.component_ids
-            )
-        )
+
+        working_components: list[int] = []
+        for inv_id in self._component_pool_status_tracker.get_working_components(
+            request.component_ids
+        ):
+            if not self._component_data_caches[inv_id].has_value():
+                _logger.warning(
+                    "Exclude inverter %s from distribution, because it didn't "
+                    "send any data since the application startup.",
+                    inv_id,
+                )
+            else:
+                working_components.append(inv_id)
```

Comment:
```suggestion
            if inv_id in self._component_data_caches:
                working_components.append(inv_id)
            else:
                _logger.warning(
                    "Exclude inverter %s from distribution, because it didn't "
                    "send any data since the application startup.",
                    inv_id,
                )
```